src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -83,14 +86,17 @@ public void onResultNotifiedOld(Participation participation) {
     }
 
     /**
-     * Use the given requestBody to extract the relevant information from it
+     * Use the given requestBody to extract the relevant information from it.
+     * Fetch and attach the result's feedback items to it.
      * @param participation Participation for which the build was finished
-     * @param requestBody RequestBody containing all relevant information
+     * @param requestBody RequestBody containing the build result and its feedback items
      */
     public void onResultNotifiedNew(Participation participation, Object requestBody) throws Exception {
         log.info("Received new build result (NEW) for participation " + participation.getId());
 
         Result result = continuousIntegrationService.get().onBuildCompletedNew(participation, requestBody);
+        List<Feedback> feedbackItems = feedbackService.getFeedbackForBuildResult(result);
```

Comment:
Is this code necessary? `onBuildCompletedNew` does already put the feedback into the result, also `feedbackService.getFeedbackForBuildResult` just exists as migration for old programming exercises. I will deprecate it.

Reply:
True, it's not needed.